pkg/controller/dataimportcron-controller.go

```diff
@@ -583,12 +583,11 @@ func addDataImportCronControllerWatches(mgr manager.Manager, c controller.Contro
 	); err != nil {
 		return err
 	}
-	// Watch only for DataSource deletion
 	if err := c.Watch(&source.Kind{Type: &cdiv1.DataSource{}},
 		handler.EnqueueRequestsFromMapFunc(mapToCron),
 		predicate.Funcs{
 			CreateFunc: func(event.CreateEvent) bool { return false },
-			UpdateFunc: func(event.UpdateEvent) bool { return false },
+			UpdateFunc: func(e event.UpdateEvent) bool { return getCronName(e.ObjectOld) != "" },
```

Comment:
Why `ObjectOld` instead of `ObjectNew`?

Reply:
Just for the case some dude updated the DataSource dataImportCron label, so we'll reconcile it also back to the DIC name.

Reply:
Right, fixed to `ObjectNew`, as we don't want to update a `DataSource` with no `DataImportCron` label.